cp23/hpi-fun2/cubes/executables/generator.cpp

```diff
@@ -137,18 +139,20 @@ int main(int argc, char *argv[])
 
     testcase("max_n", "max n", []
              {
-    auto coords = generate_coords(1000, 1000, 1000);
+    auto coords = generate_coords(1000, 1000, 1000, 0.3);
     int n = coords.size();
     println(n);
     for (int i = 0; i < n; i++)
         println(coords[i][0], coords[i][1], coords[i][2]); });
 
     for (int j = 0; j < 10; j++)
     {
-
         testcase("random" + to_string(j), "random", []
                  {
-            auto coords = generate_coords(100 + rnd.next(100), 100 + rnd.next(100), 100 + rnd.next(100));
+            // define threshold to be a float between 0.3 and 0.4
+            float threshold = 0.3 + rnd.next(10) / 100.0;
```

Comment:
```suggestion
            float threshold = rnd.next(0.3, 0.4);
```

Reply:
I didn't test it: how dense is 0.3 and how sparse is 0.4?

Reply:
Also please add the base coordinate and the `n` to your testcase description.

Reply:
below 0.3 it's almost always full and above 0.46 it's almost always empty. Should I take 0.46 instead of 0.4? some testcases will just be small (like n < 50)